hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1447,6 +1453,12 @@ private void update(final Collection<RegionState> regions, final long currentTim
         if (oldestRITTime < ritTime) {
           oldestRITTime = ritTime;
         }
+        if (counter < 500) { // Record 500 oldest RITs
```

Comment:
Just in case cluster is already facing some perf issues, we might want to reduce this count. Let's make this configurable.